clinica/pipelines/statistics_surface/_model.py

```diff
@@ -0,0 +1,728 @@
+            raise ValueError(
+                "The contrast must be an interaction between one continuous "
+                "variable and one categorical variable. Your contrast contains "
+                f"the following variables : {contrast_elements}"
+            )
+        idx = 0 if categorical[0] else 1
+        categorical_contrast = contrast_elements[idx]
+        continue_contrast = contrast_elements[(idx + 1) % 2]
+        group_values = np.unique(self.df[categorical_contrast])
+        built_contrast = self.df[continue_contrast] * (
+            (self.df[categorical_contrast] == group_values[0]).astype(int)
+        ) - self.df[continue_contrast] * (
+            (self.df[categorical_contrast] == group_values[1]).astype(int)
+        )
+        self.contrasts[contrast] = built_contrast
+
+    def filename_root(self, contrast: str):
+        if contrast not in self.contrasts:
+            raise ValueError(f"Unknown contrast {contrast}.")
+        return f"interaction-{contrast}_measure-{self.feature_label}_fwhm-{self.fwhm}"
+
+
+class GLMFactory:
+    """Factory class for building GLM models.
+
+    Attributes
+    ----------
+    feature_label: Label used for building output filenames.
+    """
+
+    def __init__(self, feature_label: str) -> None:
+        self.feature_label = feature_label
+
+    def create_model(
+        self,
+        glm_type: str,
+        design: str,
+        df: pd.DataFrame,
+        contrast: str,
+        **kwargs,
+    ) -> GLM:
+        """Factory method for building a GLM model instance corresponding to the
+        provided type and design matrix.
+
+        Parameters
+        ----------
+        glm_type: Type of GLM to be created. Either "correlation" or "group_comparison".
+        design: Design matrix in string format. If this contains a "*", it will be
+            interpreted as an interaction effect.
+        df: Subjects DataFrame.
+        contrast: Contrast in string format.
+        """
+        cprint(
+            msg=f"The GLM model is: {design} and the GLM type is: {glm_type}",
+            lvl="info",
+        )
+        if glm_type == "correlation":
+            return CorrelationGLM(design, df, self.feature_label, contrast, **kwargs)
+        elif glm_type == "group_comparison":
+            if "*" in design:
+                warnings.warn(
+                    "You included interaction as covariate in your model, "
+                    "please carefully check the format of your tsv files."
+                )
+                return GroupGLMWithInteraction(
+                    design, df, self.feature_label, contrast, **kwargs
+                )
+            return GroupGLM(design, df, self.feature_label, contrast, **kwargs)
+        raise ValueError(
+            f"GLM factory received an unknown GLM type: {glm_type}."
+            f"Only 'correlation' and 'group_comparison' are supported."
+        )
+
+
+class PValueResults:
```

Comment:
Good idea to implement our own record type for the GLM results. I'd suggest to turn them into a [dataclass](https://docs.python.org/3/library/dataclasses.html) as a result.

Reply:
Good suggestion. Implemented in [b558a64](https://github.com/aramis-lab/clinica/pull/669/commits/b558a6430ecebd33d5346b1a84d6e3d94abae4a7)